src/Fitting/CopulaModel.jl

```diff
@@ -0,0 +1,276 @@
+    if StatsBase.dof(M) == 0
+        return _print_empirical_summary(io, M)  # ver §2
+    end
+
+    θ  = StatsBase.coef(M)
+    nm = StatsBase.coefnames(M)
+    V  = StatsBase.vcov(M)
+
+    if V === nothing || isempty(θ)
+        println(io, "────────────────────────────────────────")
+        @printf(io, "%-14s %12s\n", "Parameter", "Estimate")
+        println(io, "────────────────────────────────────────")
+        @inbounds for (j, name) in pairs(nm)
+            @printf(io, "%-14s %12.6g\n", String(name), θ[j])
+        end
+        println(io, "────────────────────────────────────────")
+        return
+    end
+
+    se = sqrt.(LinearAlgebra.diag(V))
+    z  = θ ./ se
+    p  = 2 .* Distributions.ccdf(Distributions.Normal(), abs.(z))
+    lo, hi = StatsBase.confint(M; level=level)
+    lvl = Int(round(100*level))
+
+    println(io, "────────────────────────────────────────────────────────────────────────────────────────")
+    @printf(io, "%-14s %12s %12s %9s %10s %12s %12s\n",
+            "Parameter","Estimate","Std.Err","z-value","Pr(>|z|)","$lvl% Lo","$lvl% Hi")
+    println(io, "────────────────────────────────────────────────────────────────────────────────────────")
+    @inbounds for j in eachindex(θ)
+        @printf(io, "%-14s %12.6g %12.6g %9.3f %10.3g %12.6g %12.6g\n",
+                String(nm[j]), θ[j], se[j], z[j], p[j], lo[j], hi[j])
+    end
+    println(io, "────────────────────────────────────────────────────────────────────────────────────────")
+end
+
+function Base.show(io::IO, M::CopulaModel)
+    println(io, "$(typeof(M.result)) fitted via $(M.method)")
+
+    n  = StatsBase.nobs(M)
+    ll = Distributions.loglikelihood(M)
+    @printf(io, "Number of observations: %d\n", n)
+
+    ll0 = get(M.method_details, :null_ll, NaN)
+    if isfinite(ll0); @printf(io, "Null Loglikelihood: %.4f\n", ll0); end
+    @printf(io, "Loglikelihood: %.4f\n", ll)
+
+    k = StatsBase.dof(M)
+    if isfinite(ll0) && k > 0
+        LR = 2*(ll - ll0)
+        χ² = Distributions.Chisq(k)
+        p  = Distributions.ccdf(χ², LR)
+        @printf(io, "LR Test: %.2f ∼ χ²(%d) ⟹  Pr > χ² = %.4g\n", LR, k, p)
+    end
+
+    aic = StatsBase.aic(M); bic = StatsBase.bic(M)
+    @printf(io, "AIC: %.3f   BIC: %.3f\n", aic, bic)
+
+    if isfinite(M.elapsed_sec) || M.iterations != 0 || M.converged != true
+        conv = M.converged ? "true" : "false"
+        it   = M.iterations
+        tsec = isfinite(M.elapsed_sec) ? @sprintf("%.3fs", M.elapsed_sec) : "NA"
+        println(io, "Converged: $(conv)   Iterations: $(it)   Elapsed: $(tsec)")
+    end
+
+    _print_coeftable(io, M)  # (sin cambios)
+end
+
+_default_method(::Type{<:ArchimedeanCopula})  = :mle
+# ====================== Dispatch ======================
+@inline _select_method(::Type{T}, method::Symbol) where {T<:Copula} =
+    (method === :auto) ? _default_method(T) : method
+
+@inline function _dispatch_fit(::Type{T}, U::AbstractMatrix, m::Symbol; kwargs...) where {T<:Copula}
```

Comment:
This looks like you are reinventing dispatch... Please dont ;) Simply pass the method as `Val{m}()`, and then you can write functions like : 

```julia
_fit(::ArchimedeanCopula, ::AbstractMatrix, ::Val{:mle}) = ...
_fit(::ArchimedeanCopula, ::AbstractMatrix, ::Val{:itau}) = ...
_fit(::ArchimedeanCopula, ::AbstractMatrix, ::Val{:irho}) = ...
```